src/account.ts

```diff
@@ -36,6 +36,16 @@ export interface AccountState {
     amount: string;
     staked: string;
     code_hash: string;
+    storage_usage: number;
+    locked: string;
+    balance: AccountBalance;
```

Comment:
why do we have both `locked` and `staked` here? What does `balance` refer to? how is it different from `amount`?

Reply:
> why do we have both locked and staked here?

we use `locked` to calculate total balance (`amount` + `locked`), `staked` is not used, it looks like both `staked` and `accountId` are not fetched from RPC, and can be removed. 

> What does balance refer to? how is it different from amount?

`balance` contains calculated: total balance, state staked, staked and available balance. `amount` is fetched from RPC and it's used to calculate total balance.